github/internals/entities.py

```diff
@@ -0,0 +1,799 @@
+
+    if context is not None:
+        with sentry.context:
+            sentry.context.merge(context)
+
+    try:
+        sentry.captureException()
+    finally:
+        sentry.context.clear()
+
+
+class CIEnum(Enum):
+    """Ordinary enum with a fabric"""
+    @classmethod
+    def from_str(cls, s):
+        """Fabric of CIEnum"""
+        return cls._value2member_map_.get(s)
+
+
+@unique
+class Label(CIEnum):
+    """Enum which represents the label of a pull request"""
+    ACK = "ack"
+    BLACKLIST = "blacklisted"
+    POSTPONE = "postponed"
+    RERUN = "re-run"
+    REBASE = "needs rebase"
+    PRIORITIZE = "prioritize"
+
+
+@unique
+class State(CIEnum):
+    """Enum which represents the status of a commit"""
+    PENDING = "PENDING"
+    FAILURE = "FAILURE"
+    SUCCESS = "SUCCESS"
+    ERROR = "ERROR"
+
+
+class Map(dict):
+    """dot notation access to dictionary attributes"""
+    __getattr__ = dict.get
+    __setattr__ = dict.__setitem__
+    __delattr__ = dict.__delitem__
+
+
+class RateLimit(object):
+    """Represents the GitHub API rate limit"""
+    def __init__(self, limit, remaining, reset_at):
+        self.limit = limit
+        self.remaining = remaining
+        self.reset_at = reset_at
+
+    @property
+    def available(self):
+        return self.remaining >= EPHEMERAL_LIMIT
+
+    @staticmethod
+    def from_dict(data_dict):
+        """Fabric of RateLimit"""
+        return RateLimit(
+            limit=data_dict["limit"],
+            remaining=data_dict["remaining"],
+            reset_at=data_dict["reset"]
+        )
+
+    @staticmethod
+    def __get_limit(which=None):
+        """Calls GitHub API and returns RateLimit instance"""
+        if which not in ["graphql", "core"]:
+            return None
+
+        return RateLimit.from_dict(
+            World.state.github_api.rate_limit()["resources"][which]
+        )
+
+    @staticmethod
+    def get_rest_limit():
+        """Calls GitHub API and returns RateLimit instance for REST"""
+        return RateLimit.__get_limit("core")
+
+    @staticmethod
+    def get_graphql_limit():
+        """Calls GitHub API and returns RateLimit instance for GraphQL"""
+        return RateLimit.__get_limit("graphql")
+
+
+class World(object):
+    """Represents the outside world state"""
+    state = Map()
+
+    def __init__(
+        self, graphql_request, github_api, session, repo_owner,
+        repo_name, runner_id, tasks_path, whitelist
+    ):
+        self.state.available_resources = AvailableResources()
+        self.state.graphql_request = graphql_request
+        self.state.github_api = github_api
+        self.state.session = session
+        self.state.repo_owner = repo_owner
+        self.state.repo_name = repo_name
+        self.state.runner_id = runner_id
+        self.state.tasks_path = tasks_path
+        self.state.whitelist = whitelist
+        self.state.instance = self
+
+    @staticmethod
+    def __check_limit(which=None):
```

Comment:
Maybe we could change `which` for `protocol`

Reply:
Changed to `resource`